swim/memberlist_test.go

```diff
@@ -200,6 +200,28 @@ func (s *MemberlistTestSuite) TestMultipleUpdates() {
 	s.Equal(Leave, member.Status, "expected member to be leave")
 }
 
+func (s *MemberlistTestSuite) TestUpdateTriggersReincarnation() {
+	applied := s.m.Update([]Change{
+		Change{
+			Source:            "192.0.2.1:1234",
```

Comment:
Like in the PR for Go, can we make sure this value does not equal `s.m.local.Address` to make sure the test is correct?